lib/elixir/lib/file.ex

```diff
@@ -1340,6 +1355,22 @@ defmodule File do
   defp normalize_modes([], true), do: [:binary]
   defp normalize_modes([], false), do: []
 
+  defp normalize_write_modes(modes) do
+    normalize_modes([:write] ++ modes, true)
+    |> set_raw_unless_encoding_specified
+  end
+
+  # Using raw mode means we get the performance benefit of using `writev`.
+  # However, Erlang will raise a badarg error if a file encoding is specified
+  # and raw mode is also requested.
+  # http://erlang.org/doc/man/file.html#open-2
+  defp set_raw_unless_encoding_specified(modes) do
+    case Keyword.has_key?(modes, :encoding) do
```

Comment:
I debated about that. This was one line shorter and seemed readable, but I'm happy to change it if you prefer.